### DISCUSSION TITLE
SNP compiling - MPQ & Dat File

### DISCUSSION THREAD

**Discussion by impromptu1583** [bwapi/bwapi#909] [Q&A]
Hello,

I'm trying to create a new SNP to provide a modern networking option for SC/BW (specifically using valve's gamenetworkingsockets). I've been using your DirectIP snp as an example (And will of course provide full credit and will comply with license when I release my project).

The one thing I've been unable to figure out is how the .dat & .mpq files are generated. I was able to open the MPQ with pyMPQ and inspect the attached .dat file but the format is unclear (aside from the ASCII text of course). Is there any documentation you can point me to?

**Comment by heinermann** [comment]
Hello, I don't quite remember the format but think it's related to the CAPS structure in storm. They were just edited manually. You can probably look at the Hellfire source code as well to see how they did it officially.

**Reply by impromptu1583** [reply]
Thank you! I actually did sort it out. I used pympq to extract and alter the existing dat.
